src/fields.ts

```diff
@@ -473,6 +474,177 @@ export type LinkToMediaField<State extends FieldState = FieldState> =
+	 */
+	title: string | null;
+
+	/**
+	 * OEmbed resource author/owner name.
+	 */
+	author_name: string | null;
+	/**
+	 * OEmbed resource author/owner URL.
+	 */
+	author_url: string | null;
+
+	/**
+	 * OEmbed resource provider name.
+	 */
+	provider_name: string | null;
+	/**
+	 * OEmbed resource provider URL.
+	 */
+	provider_url: string | null;
+
+	/**
+	 * OEmbed suggested cache lifetime for the resource, in seconds.
+	 */
+	cache_age: number | null;
+
+	/**
+	 * OEmbed resource thumbnail URL.
+	 */
+	thumbnail_url: string | null;
+	/**
+	 * OEmbed resource thumbnail width.
+	 */
+	thumbnail_width: number | null;
+	/**
+	 * OEmbed resource thumbnail height.
+	 */
+	thumbnail_height: number | null;
+
+	/**
+	 * Providers may optionally include any parameters not specified in this
+	 * document (so long as they use the same key-value format) and consumers may
+	 * choose to ignore these. Consumers must ignore parameters they do not understand.
+	 *
+	 * @see oEmbed specification: {@link https://oembed.com}
+	 */
+	[key: string]: unknown;
+};
+
+/**
+ * OEmbed photo type.
+ *
+ * @see oEmbed specification: {@link https://oembed.com}
+ */
+export type PhotoOEmbed = OEmbedBase<typeof OEmbedType.Photo> & {
+	/**
+	 * OEmbed source URL of the image.
+	 */
+	url: string;
+	/**
+	 * OEmbed width in pixels of the image.
+	 */
+	width: number;
+	/**
+	 * OEmbed height in pixels of the image.
+	 */
+	height: number;
+};
+
+/**
+ * OEmbed video type.
+ *
+ * @see oEmbed specification: {@link https://oembed.com}
+ */
+export type VideoOEmbed = OEmbedBase<typeof OEmbedType.Video> & {
+	/**
+	 * OEmbed HTML required to embed a video player.
+	 */
+	html: string;
+	/**
+	 * OEmbed width in pixels required to display the HTML.
+	 */
+	width: number;
+	/**
+	 * OEmbed height in pixels required to display the HTML.
+	 */
+	height: number;
+};
+
+/**
+ * OEmbed link type.
+ *
+ * @see oEmbed specification: {@link https://oembed.com}
+ */
+export type LinkOEmbed = OEmbedBase<typeof OEmbedType.Link>;
+
+/**
+ * OEmbed rich type.
+ *
+ * @see oEmbed specification: {@link https://oembed.com}
+ */
+export type RichOEmbed = OEmbedBase<typeof OEmbedType.Rich> & {
+	/**
+	 * OEmbed HTML required to display the resource.
+	 */
+	html: string;
+	/**
+	 * OEmbed width in pixels required to display the HTML.
+	 */
+	width: number;
+	/**
+	 * OEmbed height in pixels required to display the HTML.
+	 */
+	height: number;
+};
+
+/**
+ * Any of the possible types of oEmbed response.
+ *
+ * @see oEmbed specification: {@link https://oembed.com}
+ */
+export type AnyOEmbed = PhotoOEmbed | VideoOEmbed | LinkOEmbed | RichOEmbed;
+
+/**
+ * An Embed field.
+ *
+ * @typeParam Data - Data provided by the URL's oEmbed provider.
+ * @typeParam State - State of the field which determines its shape.
+ * @see More details: {@link https://prismic.io/docs/core-concepts/embed}
+ */
+export type EmbedField<
+	Data extends AnyOEmbed = AnyOEmbed,
```

Comment:
Since we know there are times that the API omits fields (see the comment on line 509), I wonder if we should `extend` from something more basic. If a user knows exactly what kind of response they are expecting, this allows them to define just those fields and leave off the omitted ones.

That basic type would only include `type`, `embed_url`, and `html`. What do you think?

Reply:
Hmmm makes sense will update

Reply:
Refactored it so that `PhotoOEmbed`, `VideoOEmbed`, `LinkOEmbed`, `RichOEmbed`, or `AnyOEmbed` on feature mandatory fields, `OEmbedExtra` is the type containing all possible extra values.